samples/js/picasa/find-photos/src/main.js

```diff
@@ -204,6 +205,38 @@ async function getSizes(input: Object, origWidth: number, origHeight: number)
   return new Map(tuples);
 }
 
+async function getResources(input: Object, origWidth: number, origHeight: number)
+: Promise<Map<Struct, Struct>> {
+  const thumbURL = (await input.mediaQ24group.mediaQ24thumbnail
+    .get(0)).url.split('/');
+  const sizePart = thumbURL.length - 2;
+
+  const makeURL = s => {
+    thumbURL[sizePart] = `s${s}`;
+    return thumbURL.join('/');
+  };
+
+  const tuples = sizes.map(s => {
+    const r = fit(origWidth, origHeight, s, s);
+    if (r.scale > 1) {
+      // $FlowIssue: Does not understand that filter removes all null values.
+      return null;
+    }
+    return [
+      newStruct('', {width: r.width, height: r.height}),
+      newStruct('RemoteResource', {url: makeURL(s)}),
+    ];
+  }).filter(t => t);
+
+  // The original file.
+  tuples.push([
+    newStruct('', {width: origWidth, height: origHeight}),
+    newStruct('RemoteResource', {url: makeURL('d')}),
+  ]);
+
+  return new Map(tuples);
+}
```

Comment:
I think since this is so similar to getURLs it would be worth sharing the code ..